javascript/workflow/src/resources/Resource.ts

```diff
@@ -33,6 +33,10 @@ export abstract class Resource
 	private _dataPackage: DataPackage | undefined
 	private _unlistenToSources: Unsubscribe | undefined
 
+	public override defaultName(): string {
+		return this.profile ?? 'resource'
```

Comment:
That's an interesting idea. Because of the name de-collision mechanics it may end up being flaky but I like exposing some kind of resource property or attribute to indicate that it's JSON editable.

Reply:
Back-logged as https://dev.azure.com/msresearch/Resilience/_workitems/edit/178041. My concern is that this may result in UI knowledge leaking